src/commonTest/kotlin/com/github/quillraven/fleks/SystemTest.kt

```diff
@@ -431,4 +438,97 @@ internal class SystemTest {
         val system = world.system<SystemTestEntityCreation>()
         assertEquals(1, system.numTicks)
     }
+
+
+    @Test
+    fun iterationSystemFamilyHookFlow() {
+        // this verifies that IterationSystems properly subscribe to their family's onAdd/onRemove hooks
+        // and that the order of hook calls is correct
+
+        var check: Int = 0
+
+        val world = configureWorld {
+            families {
+                val family = family { all(NameComponent) }
+                onAdd(family) { entity ->
+                    assertEquals(0, check++) // First, we expect the global add hook to be called.
+                    println("$check. Global family onAdd() called with: entity = ${entity[NameComponent].name}")
+                }
+                onRemove(family) { entity ->
+                    assertEquals(5, check++) // The global hook gets called before the systems.
+                    println("$check. Global family onRemove() called with: entity = ${entity[NameComponent].name}")
+                }
+            }
+
+            systems {
+                add(object : IteratingSystem(family { all(NameComponent) }), IteratingSystem.FamilyOnAdd, IteratingSystem.FamilyOnRemove {
```

Comment:
extract the System outside the Test (above the Test class itself for example). In my opinion a test case itself should be as short and concise as possible and it should ideally read as easily as a book. Complex object creations should not be part of a test function. They should be extracted because otherwise it adds a lot of noise.

Reply:
It's a good point that the test code should be as simple as possible. But in that case, I'm not sure it's convenient to separate the system class declaration from the test world setup. Mostly because of the check counter. Having it all together makes it very clear what and how is tested, otherwise, partially the test would be located in the system class code declared separately. Anyways, it should be much more readable when I remove unnecessary lines and split for multiple dedicated test methods.